src/controller.rs

```diff
@@ -40,6 +40,11 @@ impl Controller {
         // Start the dummy adapter.
         let dummy_adapter = DummyAdapter::new(self.sender.clone(), self.context.clone());
         dummy_adapter.start();
+
+        {
+            let mut context = self.context.lock().unwrap();
+            context.start_tunnel().unwrap();
+        }
```

Comment:
I'd prefer to keep it explicitly there in all cases, comments can go out of date.